File: 00-Foundational-Concepts/Templates-Generics-Python/template_demo.py
from __future__ import annotations
from typing import Any


def print_multiple_times(value: Any, count: int):
    """
    Print a value multiple times.

    @param <T> type to print multiple times

    @param value value to print
    @param count number of times to print
    """

    for _ in range(0, count - 1):
        print(f"{value} ", end="")

    print(f"{value}")


def main():
    print_multiple_times(7, 3)
    print()
    print_multiple_times(5.5, 2)
    print()
    print_multiple_times("Hello", 4)
    print()
    print_multiple_times("?", 20)


if __name__ == "__main__":
    main()

File: 00-Foundational-Concepts/Templates-Generics-CPP/templateDemo.cpp
#include <string>
#include <iostream>
#include <iomanip>
#include <utility>
#include <cmath>


/**
 * Print a value multiple times.
 *
 * @tparam T type to print multiple times
 *
 * @param value value to print
 * @param count number of times to print
 */
template<typename T>
void printMultipleTimes(const T& value, const int count)
{
    for (int i = 0; i < count - 1; i++) {
        std::cout << value << ' ';
    }
    std::cout << value;
}


int main(int argc, char** argv)
{
    printMultipleTimes(7, 3);
    std::cout << "\n";
    printMultipleTimes(5.5, 2);
    std::cout << "\n";
    printMultipleTimes(5.5f, 2);
    std::cout << "\n";
    printMultipleTimes("Hello", 4);
    std::cout << "\n";
    printMultipleTimes('?', 20);
    std::cout << "\n";

    return 0;
}

